person.rb

```diff
@@ -27,13 +28,9 @@ def can_use_services?
   def correct_name
     @name
   end
-end
 
-# Testing
-person = Person.new(22, 'maximilianus')
-p person.correct_name
-person = Person.new(22, 'maximilianus')
-capitalized_person = CapitalizeDecorator.new(person)
-p capitalized_person.correct_name
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-p capitalized_trimmed_person.correct_name
+  def add_rental(rental)
+    @rentals.push(rental)
+    rental.person = self
+  end
```

Comment:
- [ ] Your `add_rental` is somewhat redundant. For us to create a rental we need a person and the rental constructor automatically adds itself to the person's `rentals` array on creation. Now if we have created a rental already for a person. we don't really need to add it again here.

  The purpose of this method is for allowing the person object create a rental given only a date and book. Kindly refactor this method.

  ```
  def add_rental(date, book)
    Rental.new(date, book, self)
  end
  ```